### PR TITLE
drivers: gpio: Add SDL simulated GPIO support

### PR DESCRIPTION
**Pull request by pdgendt**
This commit adds a driver to simulate GPIO state and interrupts from the keyboard when using SDL.

~I still need to make it build in CI. Would it make sense to extend `samples/subsys/display/lvgl` with example usage?~

### CODE REVIEW

**Review by cfriedt — COMMENTED** [review]
It's quite possible that you could wrap around the existing `gpio_emul` driver for this and provide only the callbacks.

Did you already attempt that?

**Comment by pdgendt** [comment]
> It's quite possible that you could wrap around the existing `gpio_emul` driver for this and provide only the callbacks.

Thanks @cfriedt for the suggestion, I've modified the driver as suggested.

**Review by mbolivar-nordic — DISMISSED** [review]
@petejohanson not sure if this is relevant to you but FYI.

Binding looks fine.

**Comment by petejohanson** [comment]
> @petejohanson not sure if this is relevant to you but FYI.
> 
> Binding looks fine.

Interesting! Perhaps, yes. We right now do have the ability to run as a posix process for some testing we do, this might make that a tad simpler.

No issues w/ the PR, other than maybe a request for some docs for posterity?

**Comment by pdgendt** [comment]
> No issues w/ the PR, other than maybe a request for some docs for posterity?

Where would it make sense to add documentation? SDL is currently only mentioned in the `native_posix` [board support page](https://docs.zephyrproject.org/latest/boards/posix/native_posix/doc/index.html).

File: samples/subsys/display/lvgl/src/main.c [hunk 1]
```diff
@@ -54,7 +101,7 @@ void main(void)
 			lv_label_set_text(count_label, count_str);
 		}
 		lv_task_handler();
-		k_sleep(K_MSEC(10));
 		++count;
+		k_sleep(K_MSEC(10));
```
Comment (mbolivar-nordic) [inline samples/subsys/display/lvgl/src/main.c:105]: Useless diff / unrelated change, it looks like.

**Review by mbolivar-nordic — APPROVED** [review]

File: samples/subsys/display/lvgl/src/main.c (hunk 1 above)
Comment (pdgendt) [reply samples/subsys/display/lvgl/src/main.c:105]: Not really, pressing the button resets the value immediately now, otherwise it would jump to 1 after a second

Comment (mbolivar-nordic) [reply samples/subsys/display/lvgl/src/main.c:105]: Sorry I was unclear. If it's not useless diff, it's an unrelated change and the commit is doing 2 things.

Comment (pdgendt) [reply samples/subsys/display/lvgl/src/main.c:105]: I would argument that it belongs with the commit `samples: subsys: display: lvgl: Add button to reset counter` as this introduces the new functionality.
Creating a new commit feels like a fixup.

**Review by cfriedt — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]
